kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/kroxylicious/KroxyliciousConfigMapTemplates.java

```diff
@@ -71,23 +102,12 @@ private static String getRecordEncryptionConfigMap(String clusterName, String to
                       bootstrap_servers: %CLUSTER_NAME%-kafka-bootstrap.%NAMESPACE%.svc.cluster.local:9092
                     logFrames: false
                 filters:
-                - type: RecordEncryption
-                  config:
-                    kms: VaultKmsService
-                    kmsConfig:
-                      vaultTransitEngineUrl: %VAULT_TRANSIT_ENGINE_URL%
-                      vaultToken:
-                        password: %VAULT_TOKEN%
-                    selector: TemplateKekSelector
-                    selectorConfig:
-                      template: "${topicName}"
+                %FILTER_CONFIG%
                 """
                 .replace("%NAMESPACE%", Constants.KAFKA_DEFAULT_NAMESPACE)
                 .replace("%CLUSTER_NAME%", clusterName)
                 .replace("%KROXY_SERVICE_NAME%", Constants.KROXY_SERVICE_NAME)
-                .replace("%VAULT_TOKEN%", config.vaultToken().getProvidedPassword())
-                .replace("%VAULT_TRANSIT_ENGINE_URL%", config.vaultTransitEngineUrl().toString())
-                .replace("%TOPIC_NAME%", topicName);
+                .replace("%FILTER_CONFIG%", configYaml);
```

Comment:
Another one for the list of refactorings for another day. Use `.formatted` & `%s` rather than a string of replace calls.